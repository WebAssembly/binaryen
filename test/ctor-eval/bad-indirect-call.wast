(module
  (type $v (func))
  (memory 256 256)
  (data (i32.const 10) "waka waka waka waka waka")
  (table 1 1 anyfunc)
  (elem (i32.const 0) $call-indirect)
  (export "test1" $test1)
  (func $test1
    (call_indirect $v (i32.const 1)) ;; unsafe to call, out of range
    (i32.store8 (i32.const 20) (i32.const 120))
  )
  (func $call-indirect
    (i32.store8 (i32.const 40) (i32.const 67))
  )
)
