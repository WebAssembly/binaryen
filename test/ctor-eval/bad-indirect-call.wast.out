(module
 (type $v (func))
 (memory $0 256 256)
 (data (i32.const 10) "waka waka waka waka waka")
 (table $0 1 1 funcref)
 (elem (i32.const 0) $call-indirect)
 (export "test1" (func $test1))
 (func $test1 (; 0 ;) (type $v)
  (call_indirect (type $v)
   (i32.const 1)
  )
  (i32.store8
   (i32.const 20)
   (i32.const 120)
  )
 )
 (func $call-indirect (; 1 ;) (type $v)
  (i32.store8
   (i32.const 40)
   (i32.const 67)
  )
 )
)
